src/Orchard/Environment/ThreadProvider.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading;
+
+namespace Orchard.Environment {
+    public class ThreadProvider : IThreadProvider {
```

Comment:
Gah... I hate complicating design just because you need something to unit test.. but I don't have a better suggestion, so, fair point.